Tella/Data/Database/TellaDataBase.swift

```diff
@@ -675,5 +637,119 @@ extension TellaDataBase {
+            cddl(D.cTemplateEntity, D.text),
+            cddl(D.cTemplateDownloaded, D.integer),
+            cddl(D.cTemplateUpdated, D.integer),
+            cddl(D.cTemplateFavorited, D.integer),
+            cddl(D.cServerName, D.text),
+            cddl(D.cServerId, D.integer, tableName: D.tServer, referenceKey: D.cServerId)
+
+        ]
+        statementBuilder.createTable(tableName: D.tUwaziTemplate, columns: columns)
+    }
+    func getUwaziTemplateeWith(serverId: Int) throws -> CollectedTemplate? {
+        let serversDict = try statementBuilder.selectQuery(tableName: D.tUwaziTemplate,
+                                                           andCondition: [KeyValue(key: D.cServerId, value: serverId)])
+        guard let locale = serversDict.first else { return nil }
+        return try self.parseDicToObjectOf(type: CollectedTemplate.self, dic: locale)
+    }
+    func getUwaziTemplateeWith(templateId: Int) throws -> CollectedTemplate? {
+        let serversDict = try statementBuilder.selectQuery(tableName: D.tUwaziTemplate,
+                                                           andCondition: [KeyValue(key: D.cTemplateId, value: templateId)])
+        guard let locale = serversDict.first else { return nil }
+        return try self.parseDicToObjectOf(type: CollectedTemplate.self, dic: locale)
+    }
+    func getAllUwaziTemplate() throws -> [CollectedTemplate] {
+        let serversDict = try statementBuilder.selectQuery(tableName: D.tUwaziTemplate,
+                                                           andCondition: [])
+        if !serversDict.isEmpty {
+            return try self.parseDicToObjectOf(type: [CollectedTemplate].self, dic: serversDict)
+        }
+        return []
+    }
+    func addUwaziTemplateWith(template: CollectedTemplate) throws -> CollectedTemplate {
+        let id = try statementBuilder.insertInto(tableName: D.tUwaziTemplate, keyValue: [
+            KeyValue(key: D.cTemplateId, value: template.templateId),
+            KeyValue(key: D.cTemplateDownloaded, value: 1),
+            KeyValue(key: D.cTemplateUpdated, value: 1),
+            KeyValue(key: D.cTemplateFavorited, value: 0),
+            KeyValue(key: D.cServerId, value: template.serverId),
+            KeyValue(key: D.cServerName, value: template.serverName),
+            KeyValue(key: D.cTemplateEntity, value: template.entityRowString),
+        ])
+        template.id = id
+        template.isUpdated = 1
+        template.isDownloaded = 1
+        return template
+    }
+    func deleteAllUwaziTemplate() throws -> Int {
+        return try statementBuilder.deleteAll(tableNames: [D.tUwaziTemplate])
+    }
+    func deleteAllUwaziTemplateWith(templateId: String) throws {
+        statementBuilder.delete(tableName: D.tUwaziTemplate,
+                                primarykeyValue: [KeyValue(key: D.cTemplateId, value: templateId)])
+    }
+    func deleteAllUwaziTemplateWith(templateNo: Int) throws {
```

Comment:
I think this naming is not quite descriptive. We are attempting to delete only one template but the method is called `deleteAllUwaziTemplateWith`

Also it would be more descriptive to  change `templateNo` to `id` or just `templateId`